src/array.rs

```diff
@@ -137,6 +138,55 @@ pub struct Array {
     pub(crate) c_array: mlx_array,
 }
 
```

Comment:
Should we also impl `PartialEq` with `mlx_equal` and `PartialOrd` with the `ge`, `le`, `gt`, and `lt` functions?

Reply:
I really considered it... but the way these work in `mlx` is that they return arrays where each element has been compared -- AFAIU in Rust these traits return a `bool` which means that we'd change the semantics a way to make each item is in fact equal, bigger, smaller whatever.

`mlx` does have an `array_equal` which does return an array with a single bool value in it which we could maybe use, but not sure if its clear that its using this comparison. Open to opinions here.